core/src/PrognosticData.cpp

```diff
@@ -131,27 +135,33 @@ ModelState PrognosticData::getState() const
              },
         {} };
     // clang-format on
-    // Get the state from the dynamics (ice velocity). This allows the
-    // dynamics to define its own dimensions for the velocity grid.
-    localState.merge(pDynamics->getState());
-
-    // Merge in the damage field, if the dynamics uses it.
-    if (pDynamics->usesDamage()) {
-        ModelState damageState = { {
-                                       { "damage", mask(m_damage) },
-                                   },
-            {} };
-        localState.merge(damageState);
-    }
-    return localState;
+
+    // Use the dynamics values of any duplicated fields
+    ModelState state(dynamicsState);
+    state.merge(localState);
+
+    return state;
 }
 
+// Recursively gets the data from all subcomponents
 ModelState PrognosticData::getStateRecursive(const OutputSpec& os) const
 {
-    ModelState state(getState());
-    state.merge(pAtmBdy->getStateRecursive(os));
-    state.merge(iceGrowth.getStateRecursive(os));
-    state.merge(pDynamics->getStateRecursive(os));
+    ModelState state;
+    /* If allComponents is set on the OutputSpec, then for any duplicate fields, the subsystems
+     * take priority, otherwise the fields held by PrognosticData itself. Note that std::map::merge
+     * will not overwrite existing keys, so the first one that exists will survive.
```

Comment:
Interesting. I was unfamiliar with `map.merge` before reading this. If I understand correctly, it's a merge which always chooses the existing values in the case of a conflict? It's the opposite from Python's `dict.update` method.